hedera-mirror-rosetta/scripts/deploy.sh

```
@@ -0,0 +1,62 @@
+#!/usr/bin/env bash
+set -ex
+
+cd "$(dirname $0)/.."
+
+# name of the service and directories
+name=hedera-mirror-rosetta
+configdir="/opt/${name}"
+libdir="/opt/lib/${name}"
+execname="$(ls -1 ${name}-*)"
+
+if [[ ! -f "${execname}" ]]; then
+    echo "Can't find ${execname}. Aborting"
+    exit 1
+fi
+
+mkdir -p "${configdir}" "${libdir}"
+systemctl stop "${name}.service" || true
+
+if [[ ! -f "${configdir}/application.yml" ]]; then
+    echo "Fresh install of ${execname}"
+    read -p "Database hostname: " dbHost
+    read -p "Database name: " dbName
+    read -p "Rosetta user password: " dbPassword
+    read -p "Database port: " dbPort
+    read -p "Rosetta user: " dbUser
+    read -p "Rosetta api port: " apiPort
+    cat >"${configdir}/application.yml" <<EOF
+hedera:
+  mirror:
+    rosetta:
+      apiVersion: 1.4.4
```

Why are we hardcoding all these default values? Do subsequent configuration files overwrite the values from the previous? We should fix it so that it recursively extends like the other modules, if so. Then this config should just be the minimal needed.

---

I realized the `GetConfig()` in `config.go` currently fails fast if the file doesn't exist.
Since the 1st two files aren't on the filesystem there's no configs to extend.
Importer and grpc modules executable get the extension logic cause of Spring and the classpath

---

It should exist as there's a checked in `config/application.yaml`. If it's not finding that when deployed we need to address.

---

Go 1.16 will add `embed` which handles this. Other 3rd party options exists which I can be explored.
For now will bundle the default config and ship.
That's actually not a bad idea as when new configs are added there's always a reference that ships with the new code as to how configs should be populated.

---

Added a `application-deploy.yml` with placeholders. This file is added to the deployment.
When the deploy script is run if no initial config file exists a new file will be configured and copied to the `/usr/etc` location
Embedded file logic can be addressed in a separate ticket since there's some testing to be done and since a more scalable method will be added in a new version of go vs using a 3rd party package

---

Thanks. Please open a ticket to track embedded config.